### ISSUE TITLE
Could not create server TCP listening socket *:6379: listen: 提供了一个无效的参数

### ISSUE THREAD

**Issue by zjl767864079zjl** [tporadowski/redis#88]
**Describe the bug/issue**
The redis version is 5.0.9. On some computers(three so far), when start redis,raise error Could not create server TCP listening socket *:6379: listen: 提供了一个无效的参数

**Event Log**
[16016] 26 Oct 14:30:01.264 # Redis successfully installed as a service.
[11692] 26 Oct 14:30:01.876 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
[11692] 26 Oct 14:30:01.876 # Redis version=5.0.9, bits=64, commit=9414ab9b, modified=0, pid=11692, just started
[11692] 26 Oct 14:30:01.876 # Configuration loaded
[11692] 26 Oct 14:30:01.889 # Could not create server TCP listening socket *:16379: listen: 提供了一个无效的参数。

**Comment by tporadowski** [comment]
How is your `bind` directive configured? Isn't port `16379` already in use?

**Comment by zjl767864079zjl** [comment]
> How is your `bind` directive configured? Isn't port `16379` already in use?

The port 16379 is not occupied

**Comment by tporadowski** [comment]
Please see #61 and [this comment](https://github.com/tporadowski/redis/issues/61#issuecomment-636790899) as some ports may be blocked for IPv6 protocol and Redis may try to listen on both IPv4 and IPv6 (when `bind` is empty). You can check this with:
`netsh int ipv6 show excludedportrange tcp`

**Comment by zjl767864079zjl** [comment]
> Please see #61 and [this comment](https://github.com/tporadowski/redis/issues/61#issuecomment-636790899) as some ports may be blocked for IPv6 protocol and Redis may try to listen on both IPv4 and IPv6 (when `bind` is empty). You can check this with:
> `netsh int ipv6 show excludedportrange tcp`

The problem remains.

![image](https://user-images.githubusercontent.com/24262369/98063348-d0109580-1e8a-11eb-80ce-9efefffe0f32.png)

[6684] 29 Oct 11:02:42.598 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
[6684] 29 Oct 11:02:42.598 # Redis version=5.0.9, bits=64, commit=9414ab9b, modified=0, pid=6684, just started
[6684] 29 Oct 11:02:42.598 # Configuration loaded
[6684] 29 Oct 11:02:42.602 # Could not create server TCP listening socket *:16379: listen: 提供了一个无效的参数。

**Comment by zjl767864079zjl** [comment]
> > Please see #61 and [this comment](https://github.com/tporadowski/redis/issues/61#issuecomment-636790899) as some ports may be blocked for IPv6 protocol and Redis may try to listen on both IPv4 and IPv6 (when `bind` is empty). You can check this with:
> > `netsh int ipv6 show excludedportrange tcp`
> 
> The problem remains.
> 
> ![image](https://user-images.githubusercontent.com/24262369/98063348-d0109580-1e8a-11eb-80ce-9efefffe0f32.png)
> 
> [6684] 29 Oct 11:02:42.598 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
> [6684] 29 Oct 11:02:42.598 # Redis version=5.0.9, bits=64, commit=9414ab9b, modified=0, pid=6684, just started
> [6684] 29 Oct 11:02:42.598 # Configuration loaded
> [6684] 29 Oct 11:02:42.602 # Could not create server TCP listening socket *:16379: listen: 提供了一个无效的参数。

C:\smx\Microservice\package\toolkit\nosql\redis5.0.9>netsh int ipv6 show excludedportrange tcp

协议 tcp 端口排除范围

开始端口    结束端口
----------    --------
        80          80
      5011        5011
      5357        5357
      5700        5700
      8090        8090
      8884        8884
    50248      50248

* - 管理的端口排除。

**Comment by tporadowski** [comment]
Could you please attach your config here? Have you tried a different port to see if the same problem persists?

**Comment by zjl767864079zjl** [comment]
> Could you please attach your config here? Have you tried a different port to see if the same problem persists?

I tried three ports 6379 and 26379 and 36379, the error is same.

**Config**
[redis.windows-service.zip](https://github.com/tporadowski/redis/files/5486602/redis.windows-service.zip)

**Comment by zjl767864079zjl** [comment]
The problem has been resolved. Because the bind 127.0.0.1 in the configuration is annotated, some computers throw the exception.
If you want to listen all ip, modify 127.0.0.1 to 0.0.0.0

**Comment by peter-fu** [comment]
> The problem has been resolved. Because the bind 127.0.0.1 in the configuration is annotated, some computers throw the exception. If you want to listen all ip, modify 127.0.0.1 to 0.0.0.0

It still fails to start if an IPv6 address is passed to `--bind` for those computers I guess, because some of my colleagues ran into this problem while some didn't (I wasn't able to identify the exact cause, but the problematic computers seem to have multiple network interfaces on them).

I tested the following versions on one of those problematic computers, all of them have issue binding an IPv6 address via `redis-server --bind ::1`.

* Redis 5.0.14 for Windows
* Redis 5.0.10 for Windows
* Redis 5.0.9 for Windows
* Redis 4.0.14.2 for Windows

I also tried the 6.25 version which @osthink referenced in #114, and it worked. So I guess it's still a valid issue of `Redis 5.0.14 for Windows`.
